### DISCUSSION TITLE
Kubernetes Ingress Controller apply rate-limit plugin on a combination of consumer and k8 service

### DISCUSSION THREAD

**Discussion by XJShane** [Kong/kong#9449] [Help] [answered]
Dose anyone know in Kong Kubernetes Ingress Controller it is possible to apply rate-limit plugin on a combination of consumer and k8 service?
such as rate-limit-1 plugin apply to consumerX and service-1, rate-limit-2 plugin apply to consumerX and service-2, and when consumerX request service-1 and service-2 has different rate-limit policy, like request service-1 has 10 times per second and service-2 has 20 times per second.

seems like below document says it's already support, but  i test failed, when apply both consumer and  service, it will always covered by the last policy.

[https://docs.konghq.com/kubernetes-ingress-controller/latest/references/annotations/#konghqcomplugins](url)

![image](https://user-images.githubusercontent.com/14029303/191282291-2b85f740-a54b-44fd-8c1a-2abb106994b5.png)

**Comment by StarlightIbuki** [comment]
Hi. How do you configure the plugins? Could you try to send a request to admin API "/plugins" and share the result?

**Reply by XJShane** [reply]
It turns out my ingress controller version is too low, so it's not  support the combination of consumer and K8 service.

**Comment by hbagdi** [comment]
cc @kong/team-k8s

**Comment by rainest** [comment]
This works as expected for me. With [resources.yaml](https://github.com/Kong/kong/files/9924682/resources.yaml.txt) the controller creates [config.yaml](https://github.com/Kong/kong/files/9924685/config.yaml.txt).

There are two rate-limiting plugins, each attached to a service/consumer combination. Sending a request to a service with the consumer attached to that service's plugin will apply the plugin limit. Sending a request from the other consumer will not apply the limit:

```
15:52:31-0700 esenin $ curl -sv http://localhost:8080/httpbin-a/status/200 -H "apikey: red" -H "kong-debug: 1" 2>&1 | grep X-RateLimit
< X-RateLimit-Limit-Minute: 10
< X-RateLimit-Remaining-Minute: 8

15:52:40-0700 esenin $ curl -sv http://localhost:8080/httpbin-b/status/200 -H "apikey: blue" -H "kong-debug: 1" 2>&1 | grep X-RateLimit    
< X-RateLimit-Limit-Minute: 5
< X-RateLimit-Remaining-Minute: 3

15:53:11-0700 esenin $ curl -sv http://localhost:8080/httpbin-a/status/200 -H "apikey: blue" -H "kong-debug: 1" 2>&1 | grep X-RateLimit | wc -l
0

15:53:53-0700 esenin $ curl -sv http://localhost:8080/httpbin-b/status/200 -H "apikey: red" -H "kong-debug: 1" 2>&1 | grep X-RateLimit | wc -l 
0
```

```
15:54:09-0700 esenin $ curl -ks https://localhost:8888/plugins | jq ".data[] | .name,.service,.consumer,.route"   
"rate-limiting"                                                  
{
  "id": "d2da936f-5104-58c7-9e5d-b9a80a74c0ba"
}
{
  "id": "99ccd6c6-20c7-5662-b77d-5014a1034b75"
}
null
"rate-limiting"
{
  "id": "f30261fa-3bd0-5a62-8157-77dd7fa9d6ed"
}
{
  "id": "872cfbdf-b5d0-569f-b29a-0f6b27c6fd9c"
}
null
...
```
We'd need to see your configuration and test requests to diagnose further, but my best guess is that you've got another plugin defined on that service elsewhere with limits similar to one of your consumer-specific plugins.

**Reply by XJShane** [reply]
Thanks, it turns out my ingress controller version is too low, so it's not  support the combination of consumer and K8 service.

**Comment by XJShane** [comment, marked as answer]
It turns out my ingress controller version is too low, so it's not support the combination of consumer and K8 service.

So i think we can end this discussion~

**Reply by chap-dr** [reply]
Do you recall what version was too low and what version you needed to be above to get support for this? 😅
